.github/workflows/ci.yml

```diff
@@ -6,31 +6,46 @@ on: [push]
 
 jobs:
   build:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-22.04
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
       - name: Install Debian Packages
         run: |
           sudo apt update
-          sudo apt install -y flex libjson-glib-dev libxkbcommon-dev \
-            libegl1-mesa-dev libxml2-dev libxslt1-dev libyaml-dev \
-            libwayland-dev llvm-dev libclang-dev libglib2.0-dev \
-            libepoxy-dev ninja-build
-      - name: Setup Python
-        uses: actions/setup-python@v4
-        with:
-          python-version: '3.6'
+          sudo apt install -y \
+            cmake \
+            flex \
+            libclang-dev \
+            libegl1-mesa-dev \
+            libepoxy-dev \
+            libglib2.0-dev \
+            libjson-glib-dev \
+            libwayland-dev \
+            libxkbcommon-dev \
+            libxml2-dev \
+            libxslt1-dev \
+            libyaml-dev \
+            llvm-dev \
+            ninja-build \
+            python3-dev
       - name: Python Package Cache
-        uses: actions/cache@v3
+        uses: actions/cache@v4
+        id: cache
         with:
-          path: ~/.cache/pip
+          path: |
+            ~/.cache/pip/
+            .venv/
           key: ${{ runner.os }}-pip-${{ hashFiles('.github/workflows/ci.yml') }}
           restore-keys: ${{ runner.os }}-pip-
       - name: Install Python Packages
+        if: steps.cache.outputs.cache-hit != 'true'
+        env:
+          HOTDOC_BUILD_C_EXTENSION: enabled
         run: |
-          python -m pip install --upgrade pip setuptools wheel
-          HOTDOC_BUILD_C_EXTENSION=enabled pip install hotdoc meson==0.55
+          python3 -m venv --upgrade-deps .venv
+          . .venv/bin/activate
+          pip install meson==0.55 hotdoc scan-build
```

Comment:
It seems Ubuntu 22.04 features meson 0.61.

```
$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.5 LTS
Release:        22.04
Codename:       jammy
```
```bash
$ apt-cache madison meson
     meson |   0.61.2-1 | http://ports.ubuntu.com/ubuntu-ports jammy/universe arm64 Packages
```

So unless you need meson 0.55, it would be simpler to install the `meson` package that comes with the distribution.

Reply:
We want to keep testing with the oldest Meson we are supposed to support. As a matter of fact, we should apply this change to `meson.build` in a separate commit:

```diff
diff --git a/meson.build b/meson.build
index 634aa46..420cf72 100644
--- a/meson.build
+++ b/meson.build
@@ -1,5 +1,5 @@
 project('wpebackend-fdo', ['c', 'cpp'],
-	meson_version: '>=0.52.1',
+	meson_version: '>=0.55',
 	default_options: [
 		'b_ndebug=if-release',
 		'c_std=c99',
```

😉